src/Logger.cpp

```diff
@@ -0,0 +1,121 @@
+#include "Logger.hpp"
+
+namespace weblog {
+
+/**
+ * @brief PRIVATE Construct a new Logger:: Logger object.
+
+ * The constructor is private to prevent direct object creation.
+ * The LogLevel is set to silence clang-tidy warning.
+ */
+Logger::Logger()
+	: m_logLevel(LevelInfo)
+{
+}
+
+/**
+ * @brief PRIVATE Construct a new Logger:: Logger object with copy.
+ *
+ * The copy constructor is private to prevent copying the instance.
+ * @param ref The Logger object to copy.
+ */
+Logger::Logger(const Logger& ref)
+	: m_logLevel(ref.getLevel())
+{
+	(void)ref;
```

Comment:
Instead of using c style voidcasting I would recommend using c++ style static void casting

Reply:
I'm ok with this!
Changed